File: runme/mcgen/local/datan/exasrc/e1gr.c
#include <stdio.h>
#include <string.h>
#include "grsrc.h"
#include "datsrc.h"

/* Table of constant values */

static integer c__1 = 1, c__2 = 2, c__3 = 3, c__4 = 4, c__6 = 6, c__9 = 9,
               c__75 = 75, c__201 = 201;
static double c_d0 = 0., c_d1 = 1., c_d10 = 10., c_d1p35 = 1.35,
              c_d1p414 = 1.414, c_d1p5 = 1.5, c_d2 = 2., c_d3 = 3., c_d5 = 5.,
              c_dm10 = -10., c_dm4p5 = -4.5, c_dp2 = .2, c_dp25 = .25,
              c_dp3 = .3, c_dp4 = .4, c_dp5 = .5, c_dp8 = .8;

main()
{
    /* Local variables */
    static double delx;
    static integer i;
    static double x0, tx[201], ty[201];
    static char string[75];
    static integer nws;

/* identify program to user */
    printf("%s\n", "Program E1GR demonstrates use of");
    printf("%s\n", "GROPEN,GRCLSE,GROPWS,GRCLWS,GRWNCC,GRVWWC,");
    printf("%s\n", "GRWNWC,GRSTFR,GRFRAM,GRBOUN,GRSTCL,GRPLIN,");
    printf("%s\n", "GRBRPL,GRSCLX,GRSCLY,GRTXTF.");
    printf("\n");
/* ask for number of workstation */
    grnbws_();
    printf("%s\n", "Please, enter number of workstation:");
    printf("%s", "> ");
    scanf("%ld", &nws);
    (void)getchar();
/* open GRPACK */
    gropen_();
/* open workstation */
    gropws_(&nws);
/* define window in computing coordinates (CC) */
    grwncc_(&c_dm10, &c_d10, &c_d0, &c_dp5);
/* define view port in world coordinates (WC) */
    grvwwc_(&c_dp25, &c_d1p35, &c_dp2, &c_dp8);
/* define window in WC */
    grwnwc_(&c_d0, &c_d1p414, &c_d0, &c_d1);
/* define format in device coordinates */
    grstfr_(&nws, &c_d5, &c_d0);
/* draw frame (given by window in WC) */
    grfram_();
/* draw boundary (around window in CC) */
    grboun_();
/* scale on abscissa */
    (void)memset(string, ' ', 75);
    (void)strncpy(string, "x", 1);
    grsclx_(string, &c__1);
/* scale on ordinate */
    (void)strncpy(string, "f(x)", 4);
    grscly_(string, &c__4);
/* caption */
    (void)strncpy(string, "f(x)=(2&ps^2#@)^-1/2#exp(-(x-a)^2#/2&s^2#@)", 43);
    grtxtc_(&c_d1p5, string, &c__75);
/* compute points for polyline of standardized normal */
    delx = .1;
    x0 = -10.;
    for (i = 1; i <= 201; ++i) {
	tx[i - 1] = x0 + (i - 1) * delx;
	ty[i - 1] = sdstnr_(&tx[i - 1]);
    }
/* set color index */
    grstcl_(&c__6);
/* draw continuous polyline */
    grplin_(&c__201, tx, ty);
/* compute points for polyline of Gaussian with mean 2 */
/* and standard deviation 3 */
    for (i = 1; i <= 201; ++i) {
	tx[i - 1] = x0 + (i - 1) * delx;
	ty[i - 1] = sdnorm_(&tx[i - 1], &c_d2, &c_d3);
    }
/* draw broken polyline */
    grbrpl_(&c__1, &c_d0, &c__201, tx, ty);
/* compute and draw short straight continuous polyline */
    tx[0] = -8.;
    tx[1] = -5.;
    ty[0] = .4;
    ty[1] = .4;
    grplin_(&c__2, tx, ty);
/* place text to the right of short polyline */
    (void)memset(string, ' ', 75);
    (void)strncpy(string, "a=0, &s=1", 9);
    grtxtf_(&c_dm4p5, &c_dp4, &c_d1, string, &c__75);
/* compute and draw short straight broken polyline */
    ty[0] = .3f;
    ty[1] = .3f;
    grbrpl_(&c__1, &c_d0, &c__2, tx, ty);
/* place text to its right */
    (void)strncpy(string, "a=2, &s=3", 9);
    grtxtf_(&c_dm4p5, &c_dp3, &c_d1, string, &c__75);
/* close workstation */
    grclws_(&nws);
/* close GRPACK */
    grclse_();
    return 0;
} /* main */

File: runme/mcgen/local/datan/exasrc/e1gr.f
      PROGRAM E1GR
      IMPLICIT DOUBLE PRECISION (A-H,O-Z)
      DIMENSION TX(201),TY(201)
      CHARACTER*75 STRING
C identify program to user
      WRITE(*,*)' Program E1GR demonstrates use of'
      WRITE(*,*)' GROPEN,GRCLSE,GROPWS,GRCLWS,GRWNCC,GRVWWC,'
      WRITE(*,*)' GRWNWC,GRSTFR,GRFRAM,GRBOUN,GRSTCL,GRPLIN,'
      WRITE(*,*)' GRBRPL,GRSCLX,GRSCLY,GRTXTF'
      WRITE(*,*)' '
C ask for number of workstation
      CALL GRNBWS()
      WRITE(*,*)' Please, enter number of workstation:'
      WRITE(*,*)' >'
      READ(*,*)NWS
C open GRPACK
      CALL GROPEN
C open workstation
      CALL GROPWS(NWS)
C define window in computing coordinates (CC)
      CALL GRWNCC(-10.D0,10.D0,0.D0,.5D0)
C define view port in world coordinates (WC)
      CALL GRVWWC(0.25D0,1.35D0,0.2D0,.8D0)
C define window in WC
      CALL GRWNWC(0.D0,1.414D0,0.D0,1.D0)
C define format in device coordinates
      CALL GRSTFR(NWS,5.D0,0.D0)
C draw frame (given by window in WC)
      CALL GRFRAM
C draw boundary (around window in CC)
      CALL GRBOUN
C scale on abscissa
      STRING='x'
      CALL GRSCLX(STRING,1)
C scale on ordinate
      STRING='f(x)'
      CALL GRSCLY(STRING,4)
C caption
      STRING='f(x)=(2&ps^2#@)^-1/2#exp(-(x-a)^2#/2&s^2#@)'
      CALL GRTXTC(1.5D0,STRING,75)
C compute points for polyline of standardized normal
      DELX=0.1D0
      X0=-10.D0
      DO 10 I=1,201
        TX(I)=X0+(I-1)*DELX
        TY(I)=SDSTNR(TX(I))
10    CONTINUE
C set color index
      CALL GRSTCL(6)
C draw continuous polyline
      CALL GRPLIN(201,TX,TY)
C compute points for polyline of Gaussian with mean 2
C and standard deviation 3
      DO 20 I=1,201
        TX(I)=X0+(I-1)*DELX
        TY(I)=SDNORM(TX(I),2.D0,3.D0)
20    CONTINUE
C draw broken polyline
      CALL GRBRPL(1,0.D0,201,TX,TY)
C compute and draw short straight continuous polyline
      TX(1)=-8.D0
      TX(2)=-5.D0
      TY(1)=.4D0
      TY(2)=.4D0
      CALL GRPLIN(2,TX,TY)
C place text to the right of short polyline
      STRING='a=0, &s=1'
      CALL GRTXTF(-4.5D0,.4D0,1.D0,STRING,75)
C compute and draw short straight broken polyline
      TY(1)=.3
      TY(2)=.3
      CALL GRBRPL(1,0.D0,2,TX,TY)
C place text to its right
      STRING='a=2, &s=3'
      CALL GRTXTF(-4.5D0,.3D0,1.D0,STRING,75)
C close workstation
      CALL GRCLWS(NWS)
C close GRPACK
      CALL GRCLSE
      END
